.github/workflows/farshow-build.yml

```diff
@@ -0,0 +1,98 @@
+name: Farshow build
+
+on: [push]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    container:
+      image: debian:bookworm
+
+    steps:
+    - name: Install dependencies
+      run: |
+        apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -qqy \
+          clang-format \
+          clang-tidy \
+          cmake \
+          g++ \
+          git \
+          libglew-dev \
+          libglfw3-dev \
+          libopencv-dev \
+          libopengl-dev \
+          make
+
+    - name: Add repository to safe list
+      run: git config --global --add safe.directory "$GITHUB_WORKSPACE"
+
+    - name: Checkout sources
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
+
+    - name: Initialize building environment
+      run: |
+        mkdir build
+        cd build
+        cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON ..
+
+    - name: Format
```

Comment:
I guess this and next step could be wrapped in one task called `Code linting` or something like that